src/marqo/tensor_search/tensor_search.py

```diff
@@ -2648,3 +2653,14 @@ def delete_documents(config: Config, index_name: str, doc_ids: List[str]):
             document_ids=doc_ids,
         )
     )
+
+def normalize_custom_vector(embeddings):
+    embeddings_array = np.array(embeddings)
+    magnitude = np.linalg.norm(embeddings_array)
+    if magnitude != 0:
+        embeddings_array = embeddings_array / magnitude
+    else:
+        embeddings_array = embeddings_array
```

Comment:
In what situations would magnitude be 0? It might be worth testing this edge case.

Reply:
It can be zero if the embeddings list passed originally to this function contains a list of 0s. Let me add a test case for this

Reply:
Added a test case with zero vector